.github/workflows/python-versions.yml

```diff
@@ -0,0 +1,52 @@
+name: Test Python Versions
+
+on:
+  pull_request:
+    types: [opened]
+  push:
+    branches:
+      - '**'
+
+jobs:
+  python_tests:
+
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read 
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7.12", "3.10.0"]
+    
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Run make develop
+        run: make develop
+      - name: Run Linters
+        run: |
+          # Run eslint as a standalone command to generate the test report.
+          PRE_COMMIT_NO_CONCURRENCY=true DISABLE=eslint pipenv run pre-commit run --show-diff-on-failure --color=always --all-files
```

Comment:
```suggestion
          PRE_COMMIT_NO_CONCURRENCY=true pipenv run pre-commit run --show-diff-on-failure --color=always --all-files
```
I can see that the eslint - `make jslint` call is missing here.
https://github.com/streamlit/streamlit/blob/774ce9a02115bcad42149f114ecf475ba9d6ed32/.circleci/config.yml#L503-L504
but in our case, we can use pre-commit for all hooks since we're not publishing the results yet.